src/include/units/bits/common_quantity.h

```diff
@@ -90,34 +91,47 @@ using common_quantity_point_kind = std::common_type_t<QPK1, QPK2>;
 namespace std {
 
 template<units::Quantity Q1, units::QuantityEquivalentTo<Q1> Q2>
-  requires requires { typename common_type_t<typename Q1::rep, typename Q2::rep>; }
+requires requires
+{
+  typename common_type_t<typename Q1::rep, typename Q2::rep>;
+}
 struct common_type<Q1, Q2> {
   using type = units::common_quantity<Q1, Q2>;
 };
 
 template<units::QuantityPoint QP1, units::QuantityPointEquivalentTo<QP1> QP2>
-  requires requires { typename common_type_t<typename QP1::rep, typename QP2::rep>; }
-struct common_type<QP1, QP2> {
-  using type = units::quantity_point<
-    typename common_type_t<typename QP1::quantity_type, typename QP2::quantity_type>::dimension,
-    typename common_type_t<typename QP1::quantity_type, typename QP2::quantity_type>::unit,
-    typename common_type_t<typename QP1::quantity_type, typename QP2::quantity_type>::rep>;
+requires(requires { typename common_type_t<typename QP1::rep, typename QP2::rep>; } &&
+         std::is_same_v<typename QP1::origin, typename QP2::origin>) struct common_type<QP1, QP2> {
```

Comment:
`struct` keyword should not be at the same line as constraints. clang-format is still too stupid to handle this properly.